### ISSUE TITLE
Get Quarkus and third party sources to source cache for debug symbols

### ISSUE THREAD

**Issue by galderz** [quarkusio/quarkus#10094]
Part of the debug symbol generation process is creating a source cache with the java source files available. JDK, and some of the GraalVM, source code is already added to this cache. However, sources for dependencies of the application, both Quarkus and third party, are not currently exploded. This enhancement is to make sure those are exploded and they can be navigated through with a native code debugger, e.g. gdb.

The changes should be to the Quarkus build classes in order to align with the specification of debug info [here](https://github.com/oracle/graal/blob/master/substratevm/DEBUGINFO.md).

**Comment by jaikiran** [comment]
Hello @galderz,

If I understood this part doc correctly https://github.com/oracle/graal/blob/master/substratevm/DEBUGINFO.md:
```
Option DebugInfoSourceSearchPath can be repeated as many times as required to notify all the target source locations. The value passed to this option can be either an absolute or relative path. It can identify either a directory, a source jar or a source zip file.
```
then I don't think Quarkus needs to explode the source (jar) of Quarkus application and its dependencies, isn't it? I guess, Quarkus build process could just point to the existing Maven/Gradle downloaded source jar files (that reside in the local Maven filesystem cache/repo) for each of these dependencies (and if not yet downloaded, force Maven/Gradle to do so)?

**Comment by galderz** [comment]
Right, it doesn't have to explode anything, but either we need to get Quarkus to add the right search path parameters to the generation, or we put the source jars next to the normal jars.

As a first step, there should be a way to generate a `<myproject>-1.0.0-SNAPSHOT-runner-source.jar`, because that would contain the sources for all the code inside that, which is what goes into image generation. Just quickly asked @cescoffier and he said that this is likely not there right now.

Then, it's what's in the `lib` directory, which are the dependencies. There, we have 2 options:

1. First option, improve the jar production build step to put the source jars next to the jars (in the same `lib` dir). The runner jar would not reference them so they should not bloat the native image, but the native image debug symbol generation part would find those sources and would automatically add them to the source cache. I don't know what other impact adding source jars to the `lib` directory would have though.

2. Option 2, leave the `lib` dir as is, and add `DebugInfoSourceSearchPath` entries for the source jars in the local Maven repo cache. This avoids the issue in option 1.

Any preference?

**Comment by galderz** [comment]
This issue has some complexities, so I've started a [quarkus-dev discussion](https://groups.google.com/forum/#!topic/quarkus-dev/jEJsKmHQzcw). Please comment there.
